scvi/model/_condscvi.py

```diff
@@ -202,3 +203,49 @@ def train(
             plan_kwargs=plan_kwargs,
             **kwargs,
         )
+
+    @staticmethod
+    def setup_anndata(
+        adata: AnnData,
+        labels_key: Optional[str] = None,
```

Comment:
I think for a lot of these model specific setup_anndata methods, we can be more strict on what arguments are required now. E.g. in this case, the model doesn't really make sense without a labels_key. @adamgayoso

Reply:
agree, same with Solo labels key

Reply:
are you sure the model doesn't really makes sense without a labels_key? we have a test that seems to contradict that: test_solo @adamgayoso

Reply:
in the test it uses `synthetic_iid` which by default runs setup anndata with the labels key, so it's definitely necessary for this model.

Reply:
@adamgayoso sure but then it call `setup_anndata` with no labels information, which effectively overwrites whatever ``synthetic_iid`` would have registered. Is the test wrong perhaps?

Reply:
The test seems wrong if it does that. This method doesn't really do anything if labels don't exist.

Reply:
made those required